### ISSUE TITLE
Anti-kick

### ISSUE THREAD

**Issue by jaxjaxk** [DubyaDude/RubyClientPublic#98]
Leave in anti-world owner kick. Boi the literal reason we even want an anti-kick is because the world owner OFTEN is a cunt and uses it to be a cunt. 

With the world owner being able to kick you that's only one person who does not have to like you to remove you from the world. Everyone else could love you and be find with you but you still get kicked.

With a vote kick at least most people want you gone. That's clearly a more far more reasonable reason for someone to be kicked.

**Comment by Janni9009** [comment]
If the instance owner kicks you in f+ or lower acces, you've either got the wrong friends, or maybe, just maybe, you may be the "cunt" in this case.
Anti-Vote-Kick is only here because of VRChat's votekick system being a load of garbage that allows bot kicks to spam you out of the lobby for example.

**Comment by DubyaDude** [comment]
This was removed for a personal moral reason and was already debated internally.
I will not be re-adding this feature

**Comment by jaxjaxk** [comment]
1) @Janni9009 what was your original justification for non-kick client?
2) i do not appreciate you calling me a cunt just because i have legitimate reasons for having a no-kick client. do you have friends that do not get along together? if so by your own logic you are friends with cunts.
3) what you developers decided internally without even asking the users is HIGHLY not liked. Who are you? EA? 
4) will you also be removing the ability to no-clip, fly, speed hack, orbit, force clone too? because those are even more malicious than preventing someone from kicking you from a world by them self.
@DubyaDude

**Comment by DubyaDude** [comment]
I just stated this in Changelog, but here's a bit of a deeper answer to the 'why' question.

 - NoBan Removal - This did not work after the IL2CPP update, and theoretically is way to difficult to even begin to make work without your game crashing, so removing broken feature
 - NoTargetMove Removal - Been proven to cause instant bans, so removing a 'guaranteed ban' feature
 - NoKick DeBuff - A personal Moral decision. The original intent of NoKick was to anti the 'force kick' exploit. Unfortunately the way I made it also made it so you couldn't be kicked from room owners. Since then it's been a stuck feature that I as a developer was not proud of because I've seen over the years this feature being used in horrible ways. My interest in Ruby personally feels like an all time low, I want to get back into working on things and actually making something new to distract me from how shit life's been as of late, but I didn't feel proud of anything I'd make because of that still existing. I understand your frustration and I'm deeply sorry that I'm taking away a feature. It's always becomes unpopular to remove features, and I expected the push back. As I've stated many times before, for me and the other devs this is a side hobby, I don't want to feel ashamed when I work on my hobby because of a feature I decided to keep cause I was afraid of the push back.
